core/src/database/proxy_backend/soci-proxy.h

```diff
@@ -0,0 +1,263 @@
+/*
+ *
+ * soci-proxy.h
+ * ledger-core
+ *
+ * Created by Pierre Pollastri on 13/11/2018.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#ifndef SOCI_PROXY_H_INCLUDED
+#define SOCI_PROXY_H_INCLUDED
+
+// If SOCI_PROXY_DECL isn't defined yet define it now
+#ifndef SOCI_PROXY_DECL
+# define SOCI_PROXY_DECL
+#endif
+
+#include <soci-backend.h>
+#include <api/DatabaseEngine.hpp>
+#include <api/DatabaseConnectionPool.hpp>
+#include <api/DatabaseConnection.hpp>
+#include <api/DatabaseStatement.hpp>
+#include <api/DatabaseError.hpp>
+#include <api/DatabaseBlob.hpp>
+#include <api/DatabaseResultSet.hpp>
+#include <api/DatabaseValueType.hpp>
+#include <api/DatabaseColumn.hpp>
+#include <api/DatabaseResultRow.hpp>
+
+#include <cstddef>
+#include <string>
+#include <unordered_map>
+#include <utils/Either.hpp>
+#include <soci.h>
+#include <api/DatabaseRowId.hpp>
+
+#ifndef SOCI_OVERRIDE
+#   define SOCI_OVERRIDE override
+#endif
+
+// #define SOCI_PROXY_DEBUG
+#ifdef SOCI_PROXY_DEBUG
+#define SP_PRINT(p) std::cout << p << std::endl;
+#else
+#define SP_PRINT(p)
+#endif
+
+namespace soci
```

Comment:
Can you add some documentation to everything defined in this file? It’s hard to get what you’re doing by reading the method names. 😅

Reply:
I won't document here. These are SOCI interfaces, i'll just link to the documentation. Classes and names are fairly complicated to understand what this is doing and even with the documentation